src/plugins/proctracer/proctracer.cpp

```diff
@@ -0,0 +1,390 @@
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+#include <glib.h>
+#include <libvmi/libvmi.h>
+#include <json-c/json.h>
+#include "private.h"
+
+static event_response_t trace_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info){
+    trace_info *ti=(trace_info*)info->trap->data;
+    proctracer *p=(proctracer*)ti->p;
+    if (p->trace_status.find(info->regs->cr3) == p->trace_status.end())
+        return 0;
+    switch(p->format){
+        case OUTPUT_CSV:
+            printf("proctracer,trace,0x%lx,\"%s\",0x%lx\n",info->regs->cr3, ti->mod_name, ti->offset);
+            break;
+        default:
+        case OUTPUT_DEFAULT:
+            printf("[PROCTRACER] Trace point hit - CR3: 0x%lx Module: %s Offset: 0x%lx \n",info->regs->cr3, ti->mod_name, ti->offset);
+            break;
+    }
+    
+    p->trace_statistics[info->trap_pa]+=1;
+    if (p->ccov_limit > 0 && p->trace_statistics[info->trap_pa] > p->ccov_limit){
+        drakvuf_remove_trap(drakvuf,info->trap,NULL);
+    }
+    return 0;
+}
+
+static event_response_t exit_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    proctracer *p = (proctracer*)info->trap->data;
+    if (p->trace_status.find(info->regs->cr3) == p->trace_status.end())
+        return 0;
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    if (p->trace_status.find(info->regs->cr3) == p->trace_status.end()){
+        drakvuf_release_vmi(drakvuf);
+        return 0;
+    }
+
+    list<mod_info*> module_traps=p->trace_status[info->regs->cr3];
+    for (mod_info* mi: module_traps){
+        for (drakvuf_trap_t* dt: mi->traps){
+            trace_info *ti=(trace_info*)dt->data;
+            free(ti->mod_name);
+            drakvuf_remove_trap(drakvuf,dt,NULL);
```

Comment:
I think you need to free the drakvuf_trap_t structure as well, so pass `free` as the third argument. It will complain at compile time but it can be safely ignored.